internal/errors/mysql.go

```diff
@@ -50,6 +50,11 @@ var (
 	ErrRequiredMemberNotFilled = func(member string) error {
 		return Wrapf(NewErrMySQLInvalidArgumentIdentity(), "error required member not filled (member: %s)", member)
 	}
+
+	// ErrMySQLSessionNil represents a function to generate an error that the MySQL session is nil.
+	ErrMySQLSessionNil = func() error {
```

Comment:
I think this case you shouldn't use functional type for error, but if you add additional information such as below it's better
```
	db                   string
	host                 string
	port                 int
	user                 string
	pass                 string
	name                 string
	charset              string
```

Reply:
@kpango so, for example, should I implement the below code instead of this?
```go
if m.session == nil {
  return errors.Errorf("session is nil, { db: %s, host: %s, port: %d, user: %s, pass: %s, name: %s, charset: %s", m.db, m.host, m.port, m.user, m.pass, m.name, m.charset)
}
```

Reply:
```suggestion
	ErrMySQLSessionNil = func(db, host, user, name, charset string, port int) error {
		return errors.Errorf("session is nil, { db: %s, host: %s, port: %d, user: %s, name: %s, charset: %s", db, host, port, user, name, charset)
```

Reply:
or you can print them on debug log.

Reply:
I think debug log is easy.

Reply:
@kpango thx for the review, I updated the code 🙏